### PR TITLE
Merge `series/0.23` into `main`

### PR DESCRIPTION
**Pull request by danicheg**
There were a bunch of conflicts on `WebSocket` changes in core/ember. All else was not that hard to fix.

### CODE REVIEW

**Comment by danicheg** [comment]
```
[error] sbt.librarymanagement.ResolveException: Error downloading org.portable-scala:sbt-crossproject;sbtVersion=1.0;scalaVersion=2.12:1.2.0
[error]   Not found
```
@armanbilge I remember a little discussion about this one. Did you manage to examine this while working on `sbt-typelevel` or something else?

**Comment by armanbilge** [comment]
Unfortunately it's just an annoying transient error. Restarting the CI should fix it :(

Maybe some better dependency caching will help, but I haven't had a chance to work on this.

File: server/jvm/src/main/scala/org/http4s/server/middleware/GZipPlatform.scala [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+package server.middleware
+
+import cats.Functor
+import cats.effect.kernel.Sync
+import fs2.compression._
+
+private[middleware] trait GZipPlatform { this: GZip.type =>
+
+  @deprecated("Use overload with Compression constraint", "0.23.15")
+  def apply[F[_], G[_]](
```
Comment (armanbilge) [inline server/jvm/src/main/scala/org/http4s/server/middleware/GZipPlatform.scala:27]: Actually we can just remove this and the platform trait. I only kept it for binary-compatibility, not source-compatibility, so it should not cause any problem.

**Review by armanbilge — APPROVED** [review]
I will push the cleanup commit then merge. Thanks!

**Pull request merged**